src/entity/projectile/WindCharge.php

```diff
@@ -0,0 +1,144 @@
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\entity\projectile;
+
+use pocketmine\block\Door;
+use pocketmine\entity\Entity;
+use pocketmine\event\entity\EntityDamageByChildEntityEvent;
+use pocketmine\event\entity\EntityDamageByEntityEvent;
+use pocketmine\event\entity\EntityDamageEvent;
+use pocketmine\event\entity\ProjectileHitEvent;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\RayTraceResult;
+use pocketmine\math\Vector3;
+use pocketmine\network\mcpe\protocol\types\entity\EntityIds;
+use pocketmine\world\particle\WindExplosionParticle;
+use pocketmine\world\sound\WindChargeBurstSound;
+use function ceil;
+use function floor;
+use function round;
+
+class WindCharge extends Throwable{
+
+	public const RADIUS = 2.5;
+	public const DAMAGE = 1;
+
+	public static function getNetworkTypeId() : string{ return EntityIds::WIND_CHARGE_PROJECTILE; }
+
+	protected function getInitialDragMultiplier() : float{ return 0; }
+	protected function getInitialGravity() : float{ return 0; }
+
+	public function attack(EntityDamageEvent $source) : void{
+		if(!$source instanceof EntityDamageByEntityEvent || ($entity = $source->getDamager()) === null){
+				return;
+		}
+
+		$this->setOwningEntity($entity);
+
+		$this->setMotion($entity->getDirectionVector()->multiply(1.5));
+	}
+
+	protected function onHitEntity(Entity $entityHit, RayTraceResult $hitResult) : void{
+		if($this->getOwningEntity() === null) {
+			$ev = new EntityDamageByEntityEvent($this, $entityHit, EntityDamageEvent::CAUSE_PROJECTILE, self::DAMAGE);
+		} else {
+			$ev = new EntityDamageByChildEntityEvent($this->getOwningEntity(), $this, $entityHit, EntityDamageEvent::CAUSE_PROJECTILE, self::DAMAGE);
+		}
+
+		$entityHit->attack($ev);
+
+		$this->flagForDespawn();
+	}
+
+	protected function entityBaseTick(int $tickDiff = 1) : bool{
+		parent::entityBaseTick($tickDiff);
+
+		if($this->ticksLived >= 6000) {
+			$this->flagForDespawn();
+		}
+
+		return true;
+	}
+
+	protected function onHit(ProjectileHitEvent $event) : void{
+		$source = $this->getLocation();
+
+		$this->getWorld()->addSound($event->getRayTraceResult()->getHitVector(), new WindChargeBurstSound());
+		$this->getWorld()->addParticle($source, new WindExplosionParticle());
+
+		$bound = $this->getBound($this->getPosition(), self::RADIUS - 1);
+		for($x = $bound->minX; $x <= $bound->maxX; $x++) {
+			for($y = $bound->minY; $y <= $bound->maxY; $y++) {
+				for($z = $bound->minZ; $z <= $bound->maxZ; $z++) {
+					$block = $this->getWorld()->getBlockAt((int) floor($x), (int) floor($y), (int) floor($z));
+
+					// This is to avoid calling Door::onProjectileInteraction() twice due to two tiles.
+					if($block instanceof Door) {
+						if($block->isTop()) {
```

Comment:
I think it would be better to check this in `onProjectileInteraction()` in Door class.